java/src/main/java/ai/rapids/cudf/HostUDFWrapper.java

```diff
@@ -19,18 +19,56 @@
 /**
  * A wrapper around native host UDF aggregations.
  * <p>
- * This class is used to store the native handle of a host UDF aggregation and is used as
+ * This class is used to create the native handle of a host UDF aggregation and is used as
  * a proxy object to compute hash code and compare two host UDF aggregations for equality.
  * <p>
  * A new host UDF aggregation implementation must extend this class and override the
- * {@code hashCode} and {@code equals} methods for such purposes.
- * In addition, since this class implements {@code AutoCloseable}, the {@code close} method must
- * also be overridden to automatically delete the native UDF instance upon class destruction.
+ * {@code computeHashCode} and {@code isEqual} methods for such purposes.
+ *
  */
-public abstract class HostUDFWrapper implements AutoCloseable {
-  public final long udfNativeHandle;
+public abstract class HostUDFWrapper {
```

Comment:
Pardon my ignorance, but why is this not an `AutoCloseable` any more?

Reply:
Previsously `HostUDFWrapper` has a `udfNativeHandle` (UDF native instance), so it's `AutoCloseable` which is expected to release the UDF instance in it.
Now, the `udfNativeHandle` is removed, instead we create a function `createUDFInstance` to lizily create this `udfNativeHandle` and use `closeUDFInstance` it.
Now the `createUDFInstance` and `closeUDFInstance` are in the same class, and after UDF is created and used, we directly remove it, so this will reduce the possiblity of memory leak, and make the API more simple.